R/zchunk_L2233.electricity_water.R

```diff
@@ -84,406 +110,888 @@ module_water_L2233.electricity_water_DISABLED <- function(command, ...) {
+    # ^^ note: logit exponent doesn't matter here because there's no subsector competition
+
+    L2233.supplysector_info %>%
+      rename(pass.through.sector = supplysector) %>%
+      mutate(marginal.revenue.sector = "electricity",
+             marginal.revenue.market = region) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["PassThroughSector"]])) ->
+      L2233.PassThroughSector_elec_cool # --OUTPUT--
+
+    L2233.supplysector_info -> L2233.Supplysector_elec_cool # --OUTPUT--
+
+    L2233.TechMap %>%
+      select(from.supplysector, from.subsector, from.technology, to.supplysector) %>%
+      filter(from.supplysector %in% A23.globalinttech$supplysector,
+               from.subsector %in% A23.globalinttech$subsector,
+               from.technology %in% A23.globalinttech$technology) %>%
+      select(to.supplysector) %>% unique -> L2233.elec_cool_Int_supplysectors
+
+    L2233.supplysector_info %>%
+      filter(supplysector %in% L2233.elec_cool_Int_supplysectors$to.supplysector) %>%
+      mutate(electricity.reserve.margin = unique(A23.sector$electricity.reserve.margin),
+             average.grid.capacity.factor = unique(A23.sector$average.grid.capacity.factor)) %>%
+      # ^^ Margin and capacity factor assumed to be same as for electricity and elect_td_bld
+      select(one_of(LEVEL2_DATA_NAMES[["ElecReserve"]])) %>%
+      rename(average_grid.capacity.factor = average.grid.capacity.factor) ->
+      L2233.ElecReserve_elec_cool # --OUTPUT--
+
+    elec_tech_water_map %>%
+      select(to.supplysector, to.subsector) %>%
+      unique %>%
+      rename(supplysector = to.supplysector, subsector = to.subsector) ->
+      supply_sub_elec_mapping
+    # ^^ sets up simple mapping for supply to subsector for following section
+
+    L2233.supplysector_info %>%
+      left_join_error_no_match(supply_sub_elec_mapping, by = "supplysector") %>%
+      mutate(year.fillout = first(BASE_YEARS),
+             share.weight = 1) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["SubsectorShrwtFllt"]])) ->
+      L2233.SubsectorShrwtFllt_elec_cool # --OUTPUT--
+
+    L2233.SubsectorShrwtFllt_elec_cool %>%
+      mutate(logit.year.fillout = first(BASE_YEARS),
+             logit.exponent = COOLING_SYSTEM_LOGIT) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["SubsectorLogit"]])) %>%
+      mutate(logit.type = NA) -> L2233.SubsectorLogit_elec_cool # --OUTPUT--
+
+
+    ## PART 3: GLOBAL TECHNOLOGY INFORMATION RE-ASSIGNED TO NEW STRUCTURE
+
+    # Notes: (1) All global technology information from the prior electricity sector...
+    #            ...is carried over to the new locations;
+    #        (2) We use the string "sector.name" to identify tables with global technologies...
+    #            ... since not all have "GlobalTech" in their name
+
+    # Filter the L223 files list for those with "sector.name" as a column header
+    sctrFltr <- sapply(L223_data, function(x) "sector.name" %in% attributes(x)$names)
```

Comment:
Why not just `colnames(x)`? We'd like to avoid querying attributes directly. I'll change this.